lib/services/datasets.js

```diff
@@ -0,0 +1,528 @@
+'use strict';
+
+var invariant = require('invariant'),
+  geojsonhint = require('geojsonhint/object'),
+  request = require('superagent'),
+  hat = require('hat'),
+  makeService = require('../make_service'),
+  constants = require('../constants'),
+  makeURL = require('../make_url');
+
+var Datasets = module.exports = makeService('MapboxDatasets');
+
+/**
+ * To retrieve a listing of datasets for a particular account.
+ * This request requires an access token with the datasets:read scope.
+ *
+ * @param {Function} callback called with (err, datasets)
+ * @returns {undefined} nothing, calls callback
+ * @example
+ * var mapboxClient = new MapboxClient('ACCESSTOKEN');
+ * mapboxClient.listDatasets(function(err, datasets) {
+ *   console.log(datasets);
+ *   // [
+ *   //   {
+ *   //     "owner": {account},
+ *   //     "id": {dataset id},
+ *   //     "name": {dataset name},
+ *   //     "description": {dataset description},
+ *   //     "created": {timestamp},
```

Comment:
Feel free to defer this to another conversation, but I'm curious if you have thought about documenting return types/properties with documentationjs. By using placeholders in this "example", we miss an opertunity to communicate specifics about types or return formats - like the fact that the `created` property is a ISO formatted string.

Ideally:

- return types would be thoroughly documented
- examples would be pure examples and show real looking values - not place holders